dvc/fs/http.py

```diff
@@ -127,3 +129,14 @@ def fs(self):
 
     def unstrip_protocol(self, path: str) -> str:
         return path
+
+    def put_file(
+        self,
+        from_file: AnyFSPath,
+        to_info: AnyFSPath,
+        callback: Any = DEFAULT_CALLBACK,
+        **kwargs,
+    ) -> None:
+        kwargs["method"] = self.upload_method
+        self.fs.put_file(from_file, to_info, callback=callback, **kwargs)
```

Comment:
We don't want to override `FSSpecWrapper` as much as possible, so let's call `super().put_file()` here, as it already calls the appropriate underlying filesystem and takes care of invalidating the cache.